### PR TITLE
test: Drop obsolete mkdir -p calls

### PR DESCRIPTION
**Pull request by martinpitt**
Since https://github.com/cockpit-project/bots/commit/e84a589781a3fa ,
SSHConnection.write() creates the target directory by itself.

---

Follow-ups from https://github.com/cockpit-project/cockpit/pull/17645#pullrequestreview-1092152421

### CODE REVIEW

File: test/verify/check-pages [hunk 1]
```diff
@@ -563,7 +564,8 @@ OnCalendar=daily
 
         stuff = os.path.join(self.vm_tmpdir, "stuff")
         # prepare a directory for testing file autocomplete widget
-        m.execute("mkdir -p {0}/dir1; touch {0}/file1.txt".format(stuff))
+        m.execute(f"mkdir -p {stuff}/dir1")
```
Comment (allisonkarlitskaya) [inline test/verify/check-pages:567]: Confusing, but correct.

Could also be re-written as `mkdir` (without `-p`) *after* the `write()`.

File: test/verify/check-shell-multi-machine-key [hunk 2]
```diff
@@ -239,15 +239,14 @@ class TestMultiMachineKeyAuth(MachineCase):
...
-        m1.execute("mkdir -p /home/admin/.ssh")
-        m1.upload(["verify/files/ssh/id_ed25519", "verify/files/ssh/id_ed25519.pub",
-                   "verify/files/ssh/id_rsa", "verify/files/ssh/id_rsa.pub"],
-                  "/home/admin/.ssh/")
         m1.write("/home/admin/.ssh/config", """
 Host 10.111.113.2
     User user
     IdentityFile /home/admin/.ssh/id_ed25519
 """)
+        m1.upload(["verify/files/ssh/id_ed25519", "verify/files/ssh/id_ed25519.pub",
```
Comment (allisonkarlitskaya) [inline test/verify/check-shell-multi-machine-key:247]: This reordering is begging for `mkdir -p` to be added to `Machine.upload()` as well.

**Review by allisonkarlitskaya — APPROVED** [review]
Good cleanup, thanks.  Ready to go, as-is.

A nice follow-up would be the dedenting of the contents, as we were discussing.

File: test/verify/check-shell-multi-machine-key (hunk 2 above)
Comment (martinpitt) [reply test/verify/check-shell-multi-machine-key:247]: Yes, I thought the same. It's on my TODO list now, next to the dedent() thing.

File: test/verify/check-pages (hunk 1 above)
Comment (martinpitt) [reply test/verify/check-pages:567]: It was indeed confusing before, as I initially mis-read it as "make the dir for the touched file", but that's not what it did. It's at least less confusing now.

The `-p` is indeed unnecessary, but TBH I don't feel like running this again just for this.

**Pull request merged**